File: 1122.js
// 1122. Relative Sort Array
// Given two arrays arr1 and arr2, the elements of arr2 are distinct, and all elements in arr2 are also in arr1.
// Sort the elements of arr1 such that the relative ordering of items in arr1 are the same as in arr2. Elements that do not appear in arr2 should be placed at the end of arr1 in ascending order.

var relativeSortArray = function(arr1, arr2) {
    const map = {}
    for (let num of arr1) {
        map[num] = map[num] ? map[num] + 1 : 1
    }
    let i = 0
    for (let num of arr2) {
        for (let j = 0; j < map[num]; j++) {
            arr1[i] = num
            i++
        }
        delete map[num]
    }
    for (let key in map) {
        key = parseInt(key)
        for (let j = 0; j < map[key]; j++) {
            arr1[i] = key
            i++  
        }
    }
    return arr1
};

console.log(relativeSortArray([2,3,1,3,2,4,6,7,9,2,19], [2,1,4,3,9,6]))
console.log(relativeSortArray([28,6,22,8,44,17], [22,28,8,6]))

File: 1122.py
# 1122. Relative Sort Array
# Given two arrays arr1 and arr2, the elements of arr2 are distinct, and all elements in arr2 are also in arr1.
# Sort the elements of arr1 such that the relative ordering of items in arr1 are the same as in arr2. Elements that do not appear in arr2 should be placed at the end of arr1 in ascending order.

from typing import List

class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        dict2 = {}
        for num in arr2:
            dict2[num] = []
        extra = []
        for num in arr1:
            if num in dict2:
                dict2[num].append(num)
            else:
                extra.append(num)
        output = []
        for num in dict2:
            output += dict2[num]
        return output + sorted(extra)

s = Solution()
print(s.relativeSortArray([2,3,1,3,2,4,6,7,9,2,19], [2,1,4,3,9,6]))
print(s.relativeSortArray([28,6,22,8,44,17], [22,28,8,6]))